Source/Foundation/bsfCore/Mesh/BsMesh.cpp

```diff
@@ -637,7 +637,7 @@ namespace bs
 		desc.subMeshes.push_back(SubMesh(0, initialMeshData->getNumIndices(), drawOp));
 		desc.usage = usage;
 
-		SPtr<Mesh> mesh = 
+		SPtr<Mesh> mesh =
 			bs_shared_ptr<Mesh>(new (bs_alloc<Mesh>()) Mesh(initialMeshData, desc, deviceMask));
```

Comment:
Shouldn't this be "bs_new" instead of "bs_alloc" ?

Reply:
While I do agree that this could be simplified by using bs_new, I do not feel comfortable enough with the codebase yet, to authoritatively state that it should be done. I think @BearishSun should take a look at this, and tell us why we should be using `placement new` + `bs_alloc` combo at this point :smile:

Reply:
Well, the whole bsf codebase has over 40 places where `new (bs_alloc<...>)` construct is in use. I can change all of those, but prefer to reduce the amount of work in this PR to what it meant to be, a few memory related fixes :)

Reply:
Of course, I understand.

Unless @BearishSun has an objection, I don't see why moving those 40+ places to bs_new would be bad.

Reply:
I mean, in a separate pull request, possibly not even in a single one.

Reply:
Agreed :smile:

Reply:
Separate PR for this sounds good. The reason I used placement new is that `Mesh` and various other classes with `create()` methods have non-public constructors. This used to mean that `bs_new` couldn't call them, and the code failed compiling on earlier versions of MSVC. But it appears to be working now with MSVC and GCC, so I don't mind changing it (preferably to `bs_shared_ptr_new`). I'm not sure what the standard says about this however, a global method calling a private constructor.